File: Level 1/신고 결과 받기/solution.py
def solution(id_list, reports, k):
    answer = [0] * len(id_list)

    reports = set(reports)

    reported_users = {user: 0 for user in id_list}

    for report in reports:
        reported_user = report.split(" ")[1]
        reported_users[reported_user] += 1

    for report in reports:
        report_user, reported_user = report.split(" ")

        if reported_users[reported_user] >= k:
            answer[id_list.index(report_user)] += 1

    return answer

File: Level 1/신고 결과 받기/solution.js
function solution(id_list, report, k) {
  const reports = [...new Set(report)].map((report) => report.split(" "));

  const reportedUsers = {};
  const counts = {};

  reports.forEach(([reportUser, reportedUser]) => {
    reportedUsers[reportedUser] = reportedUsers[reportedUser]
      ? [...reportedUsers[reportedUser], reportUser]
      : [reportUser];
  });

  Object.values(reportedUsers).forEach((reportUsers) => {
    if (reportUsers.length >= k) {
      reportUsers.forEach((reportUser) => {
        counts[reportUser] = counts[reportUser] ? counts[reportUser] + 1 : 1;
      });
    }
  });

  return id_list.map((user) => (counts[user] ? counts[user] : 0));
}
